client/server/silicon/silicon_utils.ts

```diff
@@ -0,0 +1,21 @@
+import { EncryptedMarket, Market } from '../../graphql/graphql';
+import { Silicon } from './silicon'; // Assume this is the path to your Silicon class
+import { poseidonHashMany } from "@scure/starknet";
+
+export function verifySiliconMapping(encrypted_silicon: Silicon<EncryptedMarket>, transparent_silicon: Silicon<Market>): boolean {
+    if (encrypted_silicon.markets.length !== transparent_silicon.markets.length) return false;
+
+    for (let i = 0; i < encrypted_silicon.markets.length; i++) {
+        const encrypted_market = encrypted_silicon.markets[i];
+        const transparent_market = transparent_silicon.markets.find(m => m.game_id === encrypted_market.game_id && m.location_id === encrypted_market.location_id && m.drug_id === encrypted_market.drug_id);
+
+        if (!transparent_market) return false; // No corresponding market found in transparent_silicon
+
```

Comment:
We usually put our entire shielded structure into one hash instead of having each property be a hash because it's much fewer constraints to do a poseidon hash of size 2 than to do 2 poseidon hashes. 

Definitely okay to keep this as is now since our shielded structure is small here (vs in network states i believe it's 8 elements). Wanted to note here for future reference!